docs/rfc/0002-plugins.md

```diff
@@ -0,0 +1,184 @@
+# Plugins
+
+## Summary
+
+We have found a number of use cases where we'd like to extend the
+functionality of `jk` but don't want to link the core runtime w specific
+libraries.
+
+With a plugin system, we could implement these features outside of the
+runtime with and provide simple way to extend `jk` and experiment with ideas.
+
+The proposal is to:
+
+- Use [go-plugin](https://github.com/hashicorp/go-plugin), the plugin system
+underpinning terraform.
+- Define a first integration point to extend `jk` functionality: `Render`
+plugins.
+- Implement a helm plugin in `@jkcfg/kubernetes` that renders helm charts and
+make its Kubernetes objects available to the `jk` runtime for further
+manipulation.
+
+## Example
+
```

Comment:
It'd be nice to see a code example in this front matter -- maybe pick one of these example plugins, and show how the code wrapping it in a module would look.